File: app/src/main/java/org/magnum/logger/communication/messaging/ReceivedMessageHandler.java
package org.magnum.logger.communication.messaging;

import org.magnum.logger.Encryptor;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.telephony.SmsMessage;
import android.util.Log;

public class ReceivedMessageHandler extends BroadcastReceiver
{

	final static String TAG = "SMS";

	@Override
	public void onReceive(Context context, Intent intent)
	{
		try
		{
			long time = System.currentTimeMillis();
			String id = "";
			Bundle bundle = intent.getExtras();
			Object[] pdus = (Object[]) bundle.get("pdus");
			SmsMessage msgs[] = new SmsMessage[pdus.length];
			for (int i = 0; i < msgs.length; i++)
			{
				msgs[i] = SmsMessage.createFromPdu((byte[]) pdus[i]);
				time = msgs[i].getTimestampMillis();
				id = Encryptor.encryptPhoneNumber(msgs[i].getOriginatingAddress(), context);
				new MessageTable(context).insert(time, id, 1);
			}
			Log.d(TAG, "SMS received from " + id + " at " + time);

		}
		catch (Exception e)
		{
			Log.e(TAG, e.toString());
		}
	}
}

File: app/src/main/java/org/magnum/logger/communication/messaging/ReceivedMessageHandler.kt
package org.magnum.logger.communication.messaging

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.telephony.SmsMessage
import android.util.Log
import org.magnum.logger.Encryptor
import org.magnum.logger.Repository

class ReceivedMessageHandler : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        try {
            var time = System.currentTimeMillis()
            var id: String? = ""
            val pdus = intent.extras!!["pdus"] as Array<*>?
            val msgs = arrayOfNulls<SmsMessage>(pdus!!.size)
            for (i in msgs.indices) {
                msgs[i] = SmsMessage.createFromPdu(pdus[i] as ByteArray)
                time = msgs[i]!!.timestampMillis
                id = Encryptor.encryptPhoneNumber(msgs[i]!!.originatingAddress, context)

                Repository(context).logMessage(MessageRecord(time, id, true))
            }
            Log.d(TAG, "SMS received from $id at $time")
        } catch (e: Exception) {
            Log.e(TAG, e.toString())
        }
    }

    companion object {
        private val TAG = ReceivedMessageHandler::class.java.canonicalName
    }
}